### PR TITLE
Fixes #34572 - fix invalid redirect after host edit

### PR DESCRIPTION
**Pull request by amirfefer**
@MariaAga, could you please verify if this fix your comment here https://github.com/theforeman/foreman/pull/9047#issuecomment-1039303156

/cc @ezr-ondrej

### CODE REVIEW

File: app/views/hosts/edit.html.erb [hunk 1]
```diff
@@ -12,5 +12,6 @@
      )
   %>
 <% end %>
-
+<%# The next hidden div is a workaround for host redirect after a submit %>
+<div class='hidden' id='hidden-host-name'><%= @host.name %></div>
```
Comment (ezr-ondrej) [inline app/views/hosts/edit.html.erb:14]: We do not need a special field for this, we have a @host.name in a host[name] field :)

File: app/assets/javascripts/host_edit.js [hunk 2]
```diff
@@ -159,7 +159,6 @@ function update_capabilities(capabilities) {
 var stop_pooling;
 
 function submit_with_all_params() {
-  var host_unique_name = construct_host_name();
```
Comment (ezr-ondrej) [inline app/assets/javascripts/host_edit.js:162]: we will still need this for new hosts. We just need to figure out new vs edit.

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
This fixes edit, but it breaks new hosts again.

File: app/views/hosts/edit.html.erb (hunk 1 above)
Comment (amirfefer) [reply app/views/hosts/edit.html.erb:14]: tried that but the `host[name]` returns only the name without the domain suffix, while  the `@host.name` returns the full fqdn.

**Comment by amirfefer** [comment]
tested with edit and new host's  form

**Review by ofedoren — APPROVED** [review]
Thanks, @amirfefer, seems to be working as expected :)

UPD: Weird, but it seems I cannot reproduce the issue, for me it works even without the fix...

**Review by ezr-ondrej — APPROVED** [review]
Thanks @amirfefer and @ofedoren for the review :+1:
I've retested on my setup and it did fix the issue for me (do you have the domain names in host names enabled btw?)

**Comment by ezr-ondrej** [comment]
CP 3.2: 5133829c0
CP 3.1: 847a2468b

**Pull request merged**